File: Design Patterns於Java語言上的實習應用/japan/UNIX/src/Singleton/Q3/Singleton.java
public class Singleton {
    private static Singleton singleton = null;
    private Singleton() {
        System.out.println("インスタンスを生成しました。");
    }
    public static Singleton getInstance() {
        if (singleton == null) {
            singleton = new Singleton();
        }
        return singleton;
    }
}

File: 品味Java的21種設計模式/dp_cpp/singleton/example6/Singleton.cpp
#include "Singleton.h"

namespace cn {
namespace javass {
namespace dp {
namespace singleton {
namespace example6 {

Singleton::Singleton() {
}

Singleton Singleton::getInstance() {
    //5：直接使用已经创建好的实例
    return instance;
}
}
}
}
}
}
